django/core/urlresolvers.py

```diff
@@ -374,11 +375,19 @@ def resolve(self, path):
                         tried.append([pattern])
                 else:
                     if sub_match:
+                        # Merge captured arguments in match with submatch
                         sub_match_dict = dict(match.groupdict(), **self.default_kwargs)
                         sub_match_dict.update(sub_match.kwargs)
+
+                        # If there are *any* named groups, ignore all non-named groups.
+                        # Otherwise, pass all non-named arguments as positional arguments.
+                        sub_match_args = sub_match.args
+                        if not sub_match_dict:
+                            sub_match_args = tuple(chain(match.groups(), sub_match.args))
```

Comment:
Any reason to use `chain` here? Both `match.groups()` and `sub_match.args` are tuples, so a simple `match.groups() + sub_match.args` should suffice. Other than that, LGTM.